### PR TITLE
Add periodic jobs for conformance-image

### PR DESCRIPTION
**Pull request by dims**
- add a CI job for conformance-image using kind
- use preset-kind-volume-mounts instead of the explicit mounts
- pull job should run on all conformance related changes

Change-Id: I58a432d16441a362a447c12126b755d290ad5119

### CODE REVIEW

**Comment by dims** [comment]
/assign @BenTheElder 
/area conformance

cc @brahmaroutu

**Comment by dims** [comment]
cc @brahmaroutu

File: config/jobs/kubernetes/sig-testing/conformance-e2e.yaml [hunk 1]
```diff
@@ -35,33 +41,49 @@ presubmits:
...
+            - "--repo=sigs.k8s.io/kind=master"
+            - "--service-account=/etc/service-account/service-account.json"
+            - "--upload=gs://kubernetes-jenkins/logs"
+            - "--scenario=execute"
+            - "--"
+            # the script must run from kubernetes, but we're checking out kind
+            - "bash"
+            - "--"
+            - "-c"
+            - "cd ./../../k8s.io/kubernetes && source ./../test-infra/experiment/kind-conformance-e2e.sh"
```
Comment (stevesloka) [inline config/jobs/kubernetes/sig-testing/conformance-e2e.yaml]: If a failure happens with tests, would this print out the ones that failed?

Comment (dims) [reply config/jobs/kubernetes/sig-testing/conformance-e2e.yaml]: yes, it will. the CI grabs the e2e.log, you can get to it using `kubectl logs` too

File: config/jobs/kubernetes-security/generated-security-jobs.yaml [hunk 2]
```diff
@@ -3053,26 +3058,9 @@ presubmits:
...
-          path: /lib/modules
-          type: Directory
-        name: modules
-      - hostPath:
-          path: /sys/fs/cgroup
-          type: Directory
-        name: cgroup
       - name: ssh-security
         secret:
           defaultMode: 256
```
Comment (brahmaroutu) [inline config/jobs/kubernetes-security/generated-security-jobs.yaml]: This file should be auto-generated and should not be part of the PR I guess.

Comment (BenTheElder) [reply config/jobs/kubernetes-security/generated-security-jobs.yaml]: this one is checked in normally for :reasons:
